h2integrate/core/utilities.py

```diff
@@ -917,3 +924,51 @@ def _structured(meta_list):
         "explicit_outputs": _structured(explicit_meta),
         "implicit_outputs": _structured(implicit_meta),
     }
+
+
+def make_cache_hash_filename(config, inputs, discrete_inputs={}, cache_dir=Path("cache")):
```

Comment:
If we are going to use classes and inheritance for cacheing, I think we should make this a class method